### PR TITLE
Add documentation

### PR DESCRIPTION
**Pull request by ashwinvaidya17**
# What does this PR do?

- Add initial documentation generation scaffold. It sets-up the documentation build that combines C++ and Python docs

## TODO

- [x] Add github actions file
- [x] Organise the documentation

## To Test

```bash
doxygen Doxyfile 
cd docs
make html && python -m http.server -d build/html
```

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Comment by sovrasov** [comment]
@ashwinvaidya17 thanks for the prompt PR! Could you fetch the latest master into your branch? Does the code style failure look reasonable?

**Comment by ashwinvaidya17** [comment]
It flags the make.bat file. I didn't write it. It was autogenerated by sphinx. My next goal is to slowly add pre-commit checks across the repo.

**Comment by sovrasov** [comment]
@eugene123tw could you have a look at the docs update?

File: docs/source/python/models/ssd.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Ssd
```
Comment (eugene123tw) [inline docs/source/python/models/ssd.md:1]: @sovrasov just for reference. I think we could move `ssd.py` to `detection_model.py` in the future.

File: docs/source/guides/index.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Guides
+
+```{toctree}
+:caption: Guides
+:hidden:
+
+./model-configuration
+./releasing-to-PyPI
```
Comment (eugene123tw) [inline docs/source/guides/index.md:8]: @ashwinvaidya17 @sovrasov I'm not sure if we need to include `PyPI releasing guide` to public doc. What do you think?

**Review by eugene123tw — APPROVED** [review]
@ashwinvaidya17 Thanks, Ashwin! I’ve added a few minor comments, but overall, it looks great! We can start building on and enhancing our documentation now :+1:

File: docs/source/guides/index.md (hunk 2 above)
Comment (ashwinvaidya17) [reply docs/source/guides/index.md:8]: I kept the existing docs as is. Ideally we shouldn't have this guide in the first place. Releasing to PyPI can be easily be automated with github actions.

Comment (sovrasov) [reply docs/source/guides/index.md:8]: Right, that's only for internal usage

File: docs/source/python/models/ssd.md (hunk 1 above)
Comment (sovrasov) [reply docs/source/python/models/ssd.md:1]: I see your point, but not 100% sure it makes sense to have both abstract `DetectionModel` and inherited from it `SSD` in one file. In opposite, in the future we might want to add other detection models. At the first iteration let's try to save compatibility of imports/includes

**Pull request merged**